app/src/main/kotlin/com/simplemobiletools/draw/pro/activities/MainActivity.kt

```diff
@@ -391,26 +391,44 @@ class MainActivity : SimpleActivity(), CanvasListener {
             eyeDropper.stop()
         }
 
-        val iconId = if (isEyeDropperOn) {
-            R.drawable.ic_colorize_off_vector
-        } else {
-            R.drawable.ic_colorize_on_vector
-        }
-
-        eye_dropper.setImageResource(iconId)
+        updateButtonStates()
     }
 
     private fun bucketFillClicked() {
+        if (isEraserOn) {
+            eraserClicked()
+        } else if (isEyeDropperOn) {
+            eyeDropperClicked()
+        }
+
         isBucketFillOn = !isBucketFillOn
 
-        val iconId = if (isBucketFillOn) {
-            R.drawable.ic_bucket_fill_off_vector
-        } else {
-            R.drawable.ic_bucket_fill_on_vector
+        updateButtonStates()
+        my_canvas.toggleBucketFill(isBucketFillOn)
+    }
+
+    private fun updateButtonStates() {
+        hideBrushSettings(isEyeDropperOn || isBucketFillOn)
+
+        arrayOf(
+            eraser to isEraserOn,
+            eye_dropper to isEyeDropperOn,
+            bucket_fill to isBucketFillOn
```

Comment:
this is confusing to read, simplify it please

Reply:
[done :+1:](https://github.com/SimpleMobileTools/Simple-Draw/pull/248/commits/4957e9a84eab79434df7c191cdeb7801d18524e9)